runsc/boot/filter/config/config_cgo.go

```diff
@@ -0,0 +1,47 @@
+// Copyright 2023 The gVisor Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//	http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+package config
+
+import (
+	"golang.org/x/sys/unix"
+	"gvisor.dev/gvisor/pkg/seccomp"
+)
+
+func cgoFilters() seccomp.SyscallRules {
+	return seccomp.MakeSyscallRules(map[uintptr]seccomp.SyscallRule{
+		unix.SYS_MMAP: seccomp.Or{
+			seccomp.PerArg{
+				seccomp.AnyValue{},
+				seccomp.AnyValue{},
+				seccomp.EqualTo(unix.PROT_NONE),
+				seccomp.EqualTo(
+					unix.MAP_PRIVATE |
+						unix.MAP_ANONYMOUS |
+						unix.MAP_NORESERVE),
+			},
+			seccomp.PerArg{
+				seccomp.AnyValue{},
+				seccomp.AnyValue{},
+				seccomp.AnyValue{},
+				seccomp.EqualTo(
+					unix.MAP_PRIVATE |
+						unix.MAP_ANONYMOUS |
+						unix.MAP_STACK),
+			},
+		},
+		unix.SYS_SET_ROBUST_LIST: seccomp.MatchAll{},
+		unix.SYS_CLONE3:          seccomp.MatchAll{},
```

Comment:
`clone3` is a wide API. Unfortunately its syscall signature makes it impossible to filter any flags from it. It would be nice to not open it up.

From the glibc source, it looks like it is optional and skipped if `clone3` returns `ENOSYS`:

```c
/* The internal wrapper of clone/clone2 and clone3.  Different than __clone3,
   it will align the stack if required.  If __clone3 returns -1 with ENOSYS,
   fall back to clone or clone2.  */
extern int __clone_internal (struct clone_args *__cl_args,
			     int (*__func) (void *__arg), void *__arg);
/* clone3 wrapper with a sticky check to avoid re-issuing the syscall if
   it fails with ENOSYS.  */
extern int __clone3_internal (struct clone_args *cl_args,
			      int (*func) (void *args), void *arg)
     attribute_hidden;
```

However, the way our seccomp filter rules are created do not currently have a way to say "for this syscall, return this custom error code". So please add a TODO here (use `TODO(eperot)` to assign to me) so that we can remove this clone3 seccomp rule in the future.

Reply:
Add comments for TODO. Thanks for explaining this.